src/c_string.rs

```diff
@@ -0,0 +1,552 @@
+//! A fixed capacity [`CString`](https://doc.rust-lang.org/std/ffi/struct.CString.html).
+
+use crate::{len_type::DefaultLenType, vec::Vec, CapacityError, LenType};
+use core::{
+    borrow::Borrow,
+    cmp::Ordering,
+    error::Error,
+    ffi::{c_char, CStr, FromBytesWithNulError},
+    fmt,
+    ops::Deref,
+};
+
+/// A fixed capacity [`CString`](https://doc.rust-lang.org/std/ffi/struct.CString.html).
+///
+/// It stores up to `N - 1` non-nul characters with a trailing nul terminator.
+#[derive(Clone, Hash)]
```

Comment:
Please also add a `Debug` implementation.

Reply:
The one from `std` should work as it simply uses `Deref`.

Reply:
hmmm, this code:

```rust
            panic!(
                "DEBUG: {:?}",
                &CString::<2>::from_bytes_with_nul(b"a\0").unwrap()
            );
```

results in:

```
DEBUG: CString { inner: [97, 0] }
```

Reply:
I ended up with a manual implementation for now.

Reply:
Yes, I meant a manual implementation, like the one on `std`.